### PR TITLE
[esphome] add chart

### PR DESCRIPTION
**Pull request by hall**
#### Special notes for your reviewer:

This is mostly just a copy of the `home-assistant` chart with the extra containers and prometheus service monitor removed. I've also added it as a dependency of the `home-assistant` chart and enabled [sharing the `secrets.yaml` file between the apps](https://esphome.io/guides/faq.html#how-do-i-use-my-home-assistant-secrets-yaml).

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[radarr]`)

### CODE REVIEW

**Comment by hall** [comment]
Tests are failing because the `esphome` chart doesn't exist in the repo. I'll remove the `home-assistant` chart changes and submit those in a separate PR.

File: charts/esphome/values.yaml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  size: 5Gi
+
+## Additional hass container environment variable
+## For instance to add a http_proxy
+##
+extraEnv: {}
+
+## Additional hass container environment variable from k8s secrets
+## For instance to add a password
+## can use `!env_var` in the home assistant configuration to reference these variables
```
Comment (billimek) [inline charts/esphome/values.yaml:95]: Can probably remove this line

File: charts/esphome/values.yaml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  # storageClass: "-"
+  ##
+  ## If you want to reuse an existing claim, you can pass the name of the PVC using
+  ## the existingClaim variable
+  # existingClaim: your-claim
+  ##
+  ## If you want to use a volume on the host machine instead of a PVC:
+  # hostPath: /path/to/the/config/folder
+  accessMode: ReadWriteOnce
+  size: 5Gi
```
Comment (billimek) [inline charts/esphome/values.yaml:86]: If the configuration data is relatively small, we may want to set the default size here to something smaller (maybe `1Gi` ?)

File: charts/esphome/values.yaml (hunk 1 above)
Comment (hall) [reply charts/esphome/values.yaml]: Good catch. I just tested and `!env_var` does work in an esphome configuration file (first point here says that most of the ha directives should work: https://esphome.io/guides/faq.html#tips-for-using-esphome).

File: charts/esphome/values.yaml (hunk 2 above)
Comment (hall) [reply charts/esphome/values.yaml]: The config directory also includes compiled data by default. There's a way to move that data to a different location on a per-node basis but I'm not sure if that can be done at a global level. At any rate, you're correct that it should be a small volume.

**Pull request merged**